src/main/java/com/faforever/client/leaderboard/LeaderboardService.java

```diff
@@ -41,21 +62,134 @@
   }
 
   @Cacheable(value = CacheNames.LEADERBOARD, sync = true)
-  public CompletableFuture<List<LeaderboardEntryBean>> getEntries(LeaderboardBean leaderboard) {
+  public CompletableFuture<List<LeaderboardEntryBean>> getEntriesForPlayer(PlayerBean player) {
     ElideNavigatorOnCollection<LeaderboardEntry> navigator = ElideNavigator.of(LeaderboardEntry.class).collection()
-        .setFilter(qBuilder().string("leaderboard.technicalName").eq(leaderboard.getTechnicalName())
-            .and().instant("updateTime").after(OffsetDateTime.now().minusMonths(1).toInstant(), false))
-        .addSortingRule("rating", false);
+        .setFilter(qBuilder().intNum("player.id").eq(player.getId()));
     return fafApiAccessor.getMany(navigator)
         .map(dto -> leaderboardMapper.map(dto, new CycleAvoidingMappingContext()))
         .collectList()
         .toFuture();
   }
 
-  @Cacheable(value = CacheNames.LEADERBOARD, sync = true)
-  public CompletableFuture<List<LeaderboardEntryBean>> getEntriesForPlayer(PlayerBean player) {
-    ElideNavigatorOnCollection<LeaderboardEntry> navigator = ElideNavigator.of(LeaderboardEntry.class).collection()
-        .setFilter(qBuilder().intNum("player.id").eq(player.getId()));
+  @Cacheable(value = CacheNames.LEAGUE, sync = true)
+  public CompletableFuture<List<LeagueBean>> getLeagues() {
+    ElideNavigatorOnCollection<League> navigator = ElideNavigator.of(League.class).collection();
+    return fafApiAccessor.getMany(navigator)
+        .map(dto -> leaderboardMapper.map(dto, new CycleAvoidingMappingContext()))
+        .collectList()
+        .toFuture();
+  }
+
+  @Cacheable(value = CacheNames.LEAGUE, sync = true)
+  public CompletableFuture<LeagueSeasonBean> getLatestSeason(int leagueId) {
+    ElideNavigatorOnCollection<LeagueSeason> navigator = ElideNavigator.of(LeagueSeason.class).collection()
+        .setFilter(qBuilder().intNum("league.id").eq(leagueId))
+        .addSortingRule("startDate", false);
+    return fafApiAccessor.getMany(navigator)
+        .filter(leagueSeason -> leagueSeason.getStartDate().isBefore(OffsetDateTime.now()))
+        .next()
+        .map(dto -> leaderboardMapper.map(dto, new CycleAvoidingMappingContext()))
+        .toFuture();
+  }
+
+  public CompletableFuture<Integer> getPlayerNumberInHigherDivisions(SubdivisionBean subdivisionBean) {
```

Comment:
This could maybe lead to inconsistencies when the entries in a subdivision change, because the getEntries method needs to use a different cache and we could get a problem with cache timeouts. I think it is still a good idea to cache it and maybe nobody really notices anyway, because you would have to switch division tabs and actively look for inconsistencies.